rust/src/main.rs

```diff
@@ -32,9 +48,16 @@ fn setup_logging() -> Result<(), fern::InitError> {
             _ => log::LevelFilter::Info,
         },
         // Default to info
-        Err(_) => log::LevelFilter::Info
+        Err(_) => log::LevelFilter::Info,
     };
 
+    let logging_directory = get_logging_directory()?;
+    // Create the logging directory
+    fs::create_dir_all(&logging_directory)?;
```

Comment:
Failure to create the log directory (as with failure to find a log directory) is not reason enough for us to exit; so instead of the `?` here, we should check the error, print a warning with `eprintln`, and then continue to run.